tv_apps/smart-home/js/edit.js

```diff
@@ -179,13 +279,55 @@
     },
 
     addNewFolder: function() {
-      this.cardManager.insertNewFolder({id: 'new-folder'},
+      var folder = this.cardManager.insertNewFolder({id: 'new-folder'},
         this.cardScrollable.currentIndex);
+      folder.on('card-swapped', this.onCardSwapped.bind(this));
       this.spatialNavigator.focus(this.cardScrollable);
     },
 
+    moveToFolder: function() {
+      var cardItem = this._hoveringItem;
+      var card = this.cardManager.findCardFromCardList(
+                      {cardId: cardItem.dataset.cardId});
+      this._hoveringCard = card;
+      // Remove the card from the main list
+      // this.deleteCard(this.cardScrollable,
+      //                 this.cardScrollable.getNodeFromItem(cardItem));
+      var nodeElem = this.cardScrollable.getNodeFromItem(cardItem);
+      this.cardManager.removeCard(parseInt(nodeElem.dataset.idx, 10));
+    },
+
+    onHoveringNodeRemoved: function() {
+      var folderItem = this._hoveredItem;
+      var folder = this.cardManager.findCardFromCardList(
+                    {cardId: folderItem.dataset.cardId});
+      // Align the folderScrollable with the folder,
+      // leave this in bug 1158665
+      // this.folderScrollable.setReferenceElement(folderItem);
+
+      this.cardScrollable.spatialNavigator.focusSilently(folderItem);
+
+      var card = this._hoveringCard;
+      // Add the card into the folder
+      folder.addCard(card, 0);
+
+      var folderNode = this.cardScrollable.getNodeFromItem(this._hoveredItem);
+      this._hoveredItem.classList.remove('hovered');
+      folderNode.classList.remove('hovered');
+      this._isHovering = false;
+      this._hoveringItem = null;
+      this._hoveredItem = null;
+      this._hoveringCard = null;
+      // this.spatialNavigator.add(this.folderScrollable);
+      this.currentScrollable = this.folderScrollable;
+      this.currentNode = this.folderScrollable.getNodeFromItem(
+                                            this.folderScrollable.currentItem);
+      this.spatialNavigator.focus(this.folderScrollable);
```

Comment:
Not sure if `focus` and `setHintArrow` should be called after `card-inserted` or not